### PR TITLE
Corelli functionality to fit a bank of tubes

### LINKED ISSUE #29544: Corelli Calibrate One Bank
Link to sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction issue [#109](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/109)

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work.**

- [x] function to fit a bank of tubes (and unit test)
- [x] function for the acceptance/rejection of a tube fit (and unit test)
- [x] function to assess if the number of neutron counts warrant a reliable tube fit (and unit test)
- [x] function to lo a bank selection (and unit test)
- [x] function to calibrate one bank (and unit test)
- [x] function to calibrate a bank selection (and unit test)
- [x] criterium function outputs a Workspace2D for deviation and Z-score versus tube index
- [x] function apply_calibration (can show the instrument, too)
- [x] write a test for the reviewer
- [x] document the functions' docstring
- [x] update release notes

**To test:**
First make sure that the `scripts/` directory located under the Mantid source repository is added to the list of scripts directories. Then, in the script Editor of the workbench, run the following script (should take not more than 5 minutes):
```
from corelli.calibration.bank import calibrate_banks
from corelli.calibration.utils import apply_calibration, load_banks
counts_file = '/SNS/CORELLI/shared/tmp/CORELLI_124023_counts.nxs'
load_banks(counts_file, bank_selection='10-19', output_workspace='counts')
calibrate_banks('counts', bank_selection='10-19')
apply_calibration('counts', 'calib10', output_workspace='counts_10', show_instrument=True)
```

Open the instrument view for workspace `counts`, and compare `bank10` against the instrument view of workspace `counts_10`. All tubes except  tube13 have been successfully calibrated.

Within the `acceptances` WorkspaceGroup lies workspace `acceptance10`. Plot spectrum number 3 of workspace `acceptance10` showing the Z-score for each tube. Tube 13 has a large Z-score denoting a bad calibration.

We can mask tube13 using table `mask10` located within WorkspaceGroup `masks`. Run the following line:
```
MaskDetectors('counts_10', DetectorList=mtd['mask10'].column(0))
```
Now verify in the instrument view of workspace `counts_10` that its bank10  has tube13 masked.

[Release notes](https://github.com/mantidproject/mantid/blob/8b70d155b5cecf345006a14f64da6bdcf13368c3/docs/source/release/v6.0.0/diffraction.rst#new-features)

Closes #29544

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/corelli/calibration/utils.py [hunk 1]
```diff
@@ -56,4 +125,34 @@ def calculate_tube_calibration(workspace: InputWorkspace, tube_name: str, shadow
...
+    assert AnalysisDataService.doesExist(str(workspace)), f'No worksapce {str(workspace)} found'
+    assert AnalysisDataService.doesExist(str(calibration_table)), f'No table {str(calibration_table)} found'
+    if output_workspace is None:
+        output_workspace = str(workspace) + '_calibrated'
+
+    CloneWorkspace(InputWorkspace=workspace, OutputWorkspace=output_workspace)
+    ApplyCalibration(Workspace=output_workspace, CalibrationTable=calibration_table)
+
+    if show_instrument is True and None not in (InstrumentViewPresenter, InstrumentViewPresenter):
+        instrument_presenter = QAppThreadCall(InstrumentViewPresenter)(mtd[output_workspace])
```
Comment (peterfpeterson) [inline scripts/corelli/calibration/utils.py:155]: I didn't realize this function existed. Nice!

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Very minor changes

File: scripts/corelli/calibration/utils.py (hunk 1 above)
Comment (jmborr) [reply scripts/corelli/calibration/utils.py:155]: neither did I until Andrei asked a related questions two days ago in Mantid slack :smile:

File: scripts/corelli/calibration/bank.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        criterium_pass[failure_indexes] = False
+
+    # create an analysis summary if so requested
+    if isinstance(summary, str) and len(summary) > 0:
+        success = [1 if criterium else 0 for criterium in criterium_pass]
+        x_values = list(range(1, 1 + TUBES_IN_BANK))
+        mean, std = np.mean(deviations), np.std(deviations)
+        z_scores = np.abs((deviations - mean) / std)
+        y_values = np.array([success, deviations, z_scores]).flatten()
+        workspace = CreateWorkspace(x_values, y_values, NSpec=3, OutputWorkspace=summary,
```
[the remark this answers is not in the document]
Comment (jmborr) [reply scripts/corelli/calibration/bank.py:156]: I'm confused. `EnableLogging` is not a property of algorithm `CreateWorkspace`, nor a method of `Workspace2D`. How can I disable logging, then?

[the remark this answers is not in the document]
Comment (jmborr) [reply scripts/corelli/calibration/bank.py:156]: I added the `EnableLogging=False`

**Review by peterfpeterson — APPROVED** [review]
Looks good now

**Pull request merged**